### PR TITLE
Promote existing E2E test to ensure successful job completion after initial failure to conformance

### PR DESCRIPTION
**Pull request by globervinodhn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
 /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This PR requests to promote an existing E2E test for Jobs to conformance. This test verifies that the job is completed successfully after initial failures

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
1. The E2E test takes approximately 20 seconds for execution.
2. The test is found to be non-flaky and non-disruptive in nature.

**Does this PR introduce a user-facing change?**: NONE

release-note-none

/area conformance

cc @brahmaroutu @mgdevstack

### CODE REVIEW

File: test/e2e/apps/job.go [hunk 1]
```diff
@@ -49,9 +49,20 @@ var _ = SIGDescribe("Job", func() {
 		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "failed to ensure job completion in namespace: %s", f.Namespace.Name)
 	})
 
+<<<<<<< HEAD
```
Comment (mgdevstack) [inline test/e2e/apps/job.go]: Plz fix rebasing codelines.

File: test/e2e/apps/job.go [hunk 2]
```diff
@@ -49,9 +49,20 @@ var _ = SIGDescribe("Job", func() {
...
 	// Pods sometimes fail, but eventually succeed.
 	ginkgo.It("should run a job to completion when tasks sometimes fail and are locally restarted", func() {
 		ginkgo.By("Creating a job")
+=======
+	/*
+		Release : v1.15
+		Testname: Jobs, completion after task failure
+		Description: Create a job. Explicity cause the tasks fails once intially and then they should be restarted locally.
+						After the restart ensure that the job eventually successds and reaches completion.
```
Comment (mgdevstack) [inline test/e2e/apps/job.go]: Please lint the lines appropriately. Multiline Conformance Doc would do great, as it is now but with proper lint.

Comment (mgdevstack) [reply test/e2e/apps/job.go]: What about:
Description: Create a job. Explicitly cause the tasks to fail once initially and then they should be restarted locally.
After the restart, Job MUST eventually succeed and reach to completion.

File: test/e2e/apps/job.go (hunk 1 above)
Comment (globervinodhn) [reply test/e2e/apps/job.go]: The concern addressed with recent commit

**Comment by globervinodhn** [comment]
/hold

Reason: Analysing a similar E2E test case for "JOB" and if possible will merge it in this PR and promote together.

The E2E test in question is -> "should run a job to completion when tasks sometimes fail and are not locally restarted"

**Comment by mgdevstack** [comment]
@globervinodhn, please do not update `conformance.txt` manually, neither while resolving merge conflict (Remove all changes from it and then follow the [link](https://github.com/kubernetes/kubernetes/tree/master/test/conformance)).

_Observed this based on your [test history](https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=76423)_.

**Review by mgdevstack — COMMENTED** [review]
Please do not update `conformance.txt` manually, neither while resolving merge conflict (Remove all changes from it and then follow the [link](https://github.com/kubernetes/kubernetes/tree/master/test/conformance)).

_Observed this based on your [test history](https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=76423)._

**Comment by johnbelamaric** [comment]
I am confused, this seems to remove a test and promote/rename another test?

**Comment by globervinodhn** [comment]
> I am confused, this seems to remove a test and promote/rename another test?

I have resolved the ambiguity with the recent commit, please do review.

File: test/e2e/apps/job.go [hunk 3]
```diff
@@ -83,16 +83,14 @@ var _ = SIGDescribe("Job", func() {
...
-	ginkgo.It("should run a job to completion when tasks sometimes fail and are locally restarted", func() {
+	/*
+		Release : v1.16
+		Testname: Jobs, completion after task failure
+		Description: Explicitly cause the tasks to fail once initially. After restarting, the Job MUST
+		execute to completion and should be successful.
+	*/
+	framework.ConformanceIt("should run a job to completion when tasks sometimes fail and are locally restarted", func() {
 		ginkgo.By("Creating a job")
-		// One failure, then a success, local restarts.
```
Comment (johnbelamaric) [inline test/e2e/apps/job.go]: why are you removing this comment, is it incorrect?

File: test/e2e/apps/job.go [hunk 4]
```diff
@@ -83,8 +83,13 @@ var _ = SIGDescribe("Job", func() {
 		framework.ExpectNoError(err, "failed to get PodList for job %s in namespace: %s", job.Name, f.Namespace.Name)
 	})
 
-	// Pods sometimes fail, but eventually succeed.
-	ginkgo.It("should run a job to completion when tasks sometimes fail and are locally restarted", func() {
+	/*
+		Release : v1.16
+		Testname: Jobs, completion after task failure
+		Description: Explicitly cause the tasks to fail once initially. After restarting, the Job MUST
+		execute to completion and should be successful.
```
Comment (johnbelamaric) [inline test/e2e/apps/job.go]: should be successful -> MUST be successful? Or must remove that "and should be..."?

Comment (spiffxp) [reply test/e2e/apps/job.go]: Please address this

Comment (spiffxp) [reply test/e2e/apps/job.go]: Please address this

**Review by spiffxp — CHANGES_REQUESTED** [review]

File: test/e2e/apps/job.go (hunk 4 above)
Comment (globervinodhn) [reply test/e2e/apps/job.go]: -> Removed the "and should be successful" part

**Comment by johnbelamaric** [comment]
/close

This was merged in #80550

**Pull request closed without merging**